.github/workflows/sonar-scan.yml

```diff
@@ -0,0 +1,76 @@
+name: Unit testing and static analysis
+
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+
+jobs:
+  wait_for_workflows:
+    name: Wait for workflows
+    runs-on: ubuntu-latest
+    if: always()
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.event.merge_group.head_sha }}
+
+      - name: Wait for Workflows
+        id: wait
+        uses: smartcontractkit/chainlink-github-actions/utils/wait-for-workflows@main
+        with:
+          max-timeout: "1200"
+          polling-interval: "30"
+          exclude-workflow-names: "Lint GH Workflows,e2e_tests_custom_cl"
+          exclude-workflow-ids: ""
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+        env:
+          DEBUG: "true"
+  tools:
+    name: Get tool-versions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out Code
+        uses: actions/checkout@v3
+      - name: Parse tool-versions file
+        uses: smartcontractkit/tool-versions-to-env-action@v1.0.8
+        id: tool-versions
+    outputs:
+      go-version: ${{ steps.tool-versions.outputs.golang_version }}
+      golangci-lint-version: ${{ steps.tool-versions.outputs.golangci-lint_version }}
+
+  sonarqube:
+    name: SonarQube Scan
+    needs: [wait_for_workflows]
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
+    steps:
+      - name: Fetch blame information
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        with:
+          fetch-depth: 0 # fetches all history for all tags and branches to provide more metadata for sonar reports
+      - name: Download all workflow run artifacts
+        uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
```

Comment:
When Sonar is triggered in a separate workflow, the traditional download artifact action won't upload any (see the results [here](https://github.com/smartcontractkit/chainlink-solana/actions/runs/6083905420/job/16505025707?pr=506#step:4:7)). Please, use the same approach as in starknet repo for downloading different artifacts from independent flows.
```
      - name: Download Golangci Relayer report
        uses: dawidd6/action-download-artifact@v2.27.0
        with:
          workflow: golangci-lint.yml
          workflow_conclusion: ""
          name: golangci-lint-relayer-report
          if_no_artifact_found: warn
```